src/events/handlers/post.rs

```diff
@@ -373,12 +382,17 @@ pub async fn sync_del(author_id: PubkyId, post_id: String) -> Result<(), DynErro
                 JsonAction::Decrement(1),
             )
             .await?;
-            PostStream::put_score_index_sorted_set(
-                &POST_TOTAL_ENGAGEMENT_KEY_PARTS,
-                &parent_post_key_parts,
-                ScoreAction::Decrement(1.0),
-            )
-            .await?;
+
+            // Post replies cannot be included in the total engagement index after the reply is deleted
+            // Only root posts should be included. Ensure that the parent post is the root post
+            if PostRelationships::is_root(&parent_user_id, &parent_post_id).await? {
+                PostStream::put_score_index_sorted_set(
+                    &POST_TOTAL_ENGAGEMENT_KEY_PARTS,
+                    &parent_post_key_parts,
+                    ScoreAction::Decrement(1.0),
+                )
+                .await?;
+            }
```

Comment:
Maybe this is time to do a bit of boy scouting here. We are doing the same thing in two different ways in the same file. We have `is_reply()` function that is not part of the `impl PostRelationship` , instead it is a util hanging out here by itself at the bottom of `handlers/post.rs` , but it is functionally the same (just inverse) to `is_root()`. In specific:
```
    let relationships = PostRelationships::get_by_id(&author_id, &post_id).await?;
    let is_reply = is_reply(&relationships);
```
and
```
PostRelationships::is_root(&parent_user_id, &parent_post_id).await?
```
are just 2 different ways we are now doing the exact same thing.

Ideally we want to keep only one: `is_root()` or `is_reply()`. It should be part of the `PostRelationship` impl and use `self` `is_root(&self)` instead of doing a look up into redis (because sometimes, we will already have the `post_relationship` object and we can avoid the lookup in redis).